overlord/servicestate/service_control.go

```diff
@@ -36,10 +36,11 @@ import (
 // snap if services list is empty.
 // The names of services are app names (as defined in snap yaml).
 type ServiceAction struct {
-	SnapName       string   `json:"snap-name"`
-	Action         string   `json:"action"`
-	ActionModifier string   `json:"action-modifier,omitempty"`
-	Services       []string `json:"services,omitempty"`
+	SnapName         string   `json:"snap-name"`
+	Action           string   `json:"action"`
+	ActionModifier   string   `json:"action-modifier,omitempty"`
+	Services         []string `json:"services,omitempty"`
+	ExplicitServices []string `json:"explicit-services,omitempty"`
```

Comment:
> Perhaps the field shoudl be renamed to ExplicitAppNames, or if later use only deals with services, then we may want to keep the actual snap.app.service here too.

The problem with renaming is that the structure is called ServiceAction, and all the other names (ServiceState, ServiceControl) also point to these being services. I could expand the name as you suggest, but it'd be just duplicating information already available, since we can easily compute the full service names.

One more option would be to have this as an array of indexes into the existing `Services` field, but it seems an unnecessary complication for little benefit.

Is there a reason why the `Services` field contains the full `snap.app.service` name?

Reply:
> Is there a reason why the Services field contains the full snap.app.service name? 

it's probably historical but we can't change that as we tend to avoid such changes on tasks because then they don't work in the (unlikely but possible) face of reverts

anyway it's not just a matter of field naming, I think the best is if the two fields contain the same kind of values, much easier to then check for intersections if we don't need to manipulate them, but given that we can't change Services content we should adapt ExplicitServices, see below